packages/wasm-utils/src/note/secrets.rs

```diff
@@ -0,0 +1,67 @@
+use ark_std::rand::rngs::OsRng;
+use arkworks_circuits::setup::mixer::{setup_leaf, setup_leaf_with_privates_raw};
+use arkworks_gadgets::prelude::ark_bls12_381::Fq as BlsFr;
+use arkworks_gadgets::prelude::ark_bn254::Fq as Bn254Fr;
+use arkworks_gadgets::prelude::*;
+use arkworks_utils::utils::common::Curve as ArkworksCurve;
+
+use crate::types::{Curve, OpStatusCode};
+use crate::utils::get_hash_params_x5;
+
+pub fn generate_secrets(
+	exponentiation: i8,
+	width: usize,
+	curve: Curve,
+	rng: &mut OsRng,
+) -> Result<Vec<u8>, OpStatusCode> {
+	let secrets: Vec<u8> = match (curve, exponentiation, width) {
+		(Curve::Bls381, 5, ..) => {
```

Comment:
We should also require that the leaf width is 5
```suggestion
		(Curve::Bls381, 5, 5) => {
```